src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -2929,6 +2940,16 @@ def _update_addons(cmd,  # pylint: disable=too-many-branches,too-many-statements
                 addon_profile = ManagedClusterAddonProfile(enabled=True, config={CONST_ACC_SGX_QUOTE_HELPER_ENABLED: "false"})
                 if enable_sgxquotehelper:
                     addon_profile.config[CONST_ACC_SGX_QUOTE_HELPER_ENABLED] = "true"
+            elif addon == CONST_AZURE_KEYVAULT_SECRETS_PROVIDER_ADDON_NAME:
+                if addon_profile.enabled:
+                    raise CLIError('The azure-keyvault-secrets-provider addon is already enabled for this managed cluster.\n'
```

Comment:
if the addon is enabled, can the user change the "enable-secret-rotation" flag value?

Reply:
Discussed with Michael, we should support the scenario. I will add it later.

Reply:
Done. PTAL @robbiezhang